src/zenml/orchestrators/context_utils.py

```diff
@@ -67,17 +69,28 @@ def add_pydantic_object_as_metadata_context(
         obj.__repr_name__().lower()
     )
     # Setting the name of the context
-    name = str(hash(obj.json(sort_keys=True)))
+    try:
+        name = str(hash(obj.json(sort_keys=True)))
+    except TypeError as e:
+        class_name = obj.__class__.__name__
+        logging.info(
+            "Cannot convert %s to json, generating uuid instead. Error: %s",
+            class_name,
+            e,
+        )
+        name = f"{class_name}_{uuid.uuid1()}"
     context.name.field_value.string_value = name  # type:ignore[attr-defined]
 
-    # Setting the properties of the context
+    # Setting the properties of the context depending on attribute type
     for k, v in obj.dict().items():
-        c_property = context.properties[k]  # type:ignore[attr-defined]
-        if isinstance(v, int):
-            c_property.field_value.int_value = v
-        elif isinstance(v, float):
-            c_property.field_value.double_value = v
-        elif isinstance(v, str):
-            c_property.field_value.string_value = v
-        else:
-            c_property.field_value.string_value = str(v)
+        try:
+            value = json.dumps(v)
+            c_property = context.properties[k]  # type:ignore[attr-defined]
+            c_property.field_value.string_value = value
+        except TypeError as e:
+            logging.info(
```

Comment:
I'm wondering if we should raise an error here instead: With reproducability in mind, I'm guessing we will want to restore classes (e.g. the RuntimeConfiguration) after reading their configuration from the metadata store. I'm not sure whether it makes sense to fail early as we won't be able to deliver the promised reproducability or fail once they're actually trying to run the pipeline again?

Reply:
@schustmi yeah I was hoping to get some feedback/second opinions on this. Perhaps the default behavior should be to fail fast, but in some way allow the user to set a flag overriding this behavior and skipping the faulty fields? I don't know for sure if there's a use case for that, but I do always hate restricting the user more than necessary

Reply:
I like that you that idea of overriding behavior but defaulting to raising an exception.

Reply:
Yep agreed with that, but I have no clue how we should allow this overwriting as this is pretty deep inside ZenML land. An environment variable? Or adding a specific key/value in the runtime configuration?

Reply:
> Yep agreed with that, but I have no clue how we should allow this overwriting as this is pretty deep inside ZenML land. An environment variable? Or adding a specific key/value in the runtime configuration?
@schustmi Essentially my thought now was to use a key in the runtime configuration. This way it's opt-in if we want to document it, but works just fine when not specified